src/store/search-store.js

```diff
@@ -166,10 +165,25 @@ const actions = ImageService => ({
         );
       })
       .catch((error) => {
-        commit(FETCH_IMAGES_ERROR);
-        throw new Error(error);
+        dispatch(HANDLE_IMAGE_ERROR, error);
       });
   },
+  [HANDLE_IMAGE_ERROR]({ commit }, error) {
+    if (error.response) {
+      if (error.response.status === 500) {
+        commit(FETCH_IMAGES_ERROR, { errorMsg: 'There was a problem with our servers' });
+      }
+      else if (error.response.status === 404) {
+        commit(FETCH_IMAGES_ERROR, { errorMsg: 'No images were found for this image' });
```

Comment:
The API doesn't return a 404 when there are no results. It returns a 200 with a result count of 0. I think you can remove this and handle that differently